File: spring-example-orca/src/main/java/io/armory/plugin/example/spring/services/OverrideService.java
package io.armory.plugin.example.spring.services;
import com.netflix.spinnaker.fiat.model.resources.Role;
import com.netflix.spinnaker.fiat.roles.file.FileBasedUserRolesProvider;
import com.netflix.spinnaker.fiat.permissions.ExternalUser;

import java.util.*;

public class OverrideService extends FileBasedUserRolesProvider {

    @Override
    public List<Role> loadRoles(ExternalUser user) {
        List<Role> roles = super.loadRoles(user);
        if(roles.isEmpty()){
            roles = Collections.singletonList(new Role("public"));
        }
        else{
            roles.add(new Role("public"));
        }
//        roles.isEmpty()? Collections.singletonList(new Role("public")) : roles
        return roles;
    }


}

File: spring-example-orca/src/main/kotlin/io/armory/plugin/example/spring/services/OverrideService.kt
package io.armory.plugin.example.spring.services

import com.netflix.spinnaker.fiat.model.resources.Role
import com.netflix.spinnaker.fiat.roles.file.FileBasedUserRolesProvider
import com.netflix.spinnaker.fiat.permissions.ExternalUser

//import org.pf4j.PluginManager
//import java.io.IOException


class OverrideService() { // : FileBasedUserRolesProvider()
    override fun loadRoles(user: ExternalUser): List<Role> {
        try {
            return Optional.ofNullable(parse().get(user.getId()))
                    .orElse(listOf(Role("public")))
        } catch (io: IOException) {
            log.error("Couldn't load roles for user " + user.getId().toString() + " from file", io)
        }
        return listOf<Role>(Role("public"))
    }
}